src/map/script.c

```diff
@@ -9133,6 +9133,92 @@ static BUILDIN(grouprandomitem)
 	return true;
 }
 
+/*==========================================
+ * utils for item group
+ *------------------------------------------*/
+static bool utils_itemgroup_remove_duplicate(struct item_group *group)
+{
+	int n = group->qty;
+	int *arr = group->nameid;
+	int i, j, k;
+
+	// Remove duplicates
+	for (i = 0; i < n; i++) {
+		for (j = i + 1; j < n; j++) {
+			if (arr[i] == arr[j]) {
+				// Shift elements to the left
+				for (k = j; k < n - 1; k++) {
+					arr[k] = arr[k + 1];
+				}
+				n--;
+				j--;
+			}
+		}
+	}
+
+	// Create a new array without duplicates
+	int *newArr = malloc(n * sizeof(int));
+	if (newArr == NULL) {
+		// Handle memory allocation error
+		return false;
+	}
+
+	for (i = 0; i < n; i++) {
+		newArr[i] = arr[i];
+	}
+
+	group->qty = n;
+	group->nameid = newArr;
+
+	return true;
+}
+
+/*==========================================
+ * getitemgroupitems <reference array of item_id>, <item_group_id>;
+ *------------------------------------------*/
+static BUILDIN(getitemgroupitems)
+{
+	struct script_data* data = script_getdata(st, 2);
+
+	if (!data_isreference(data)) {
+		ShowError("buildin_getitemgroupitems: not a variable\n");
+		script->reportdata(data);
+		script_pushnil(st);
+		st->state = END;
+		return false;// not a variable
+	}
+
+	struct item_data *item = itemdb->name2id(script_getstr(st, 3));
```

Comment:
I'm not a fan of using strings where you could use integers. Can you make this function accept an integer instead? (i.e. the constant `Old_Card_Album` instead of the string `"Old_Card_Album"`).

If you think it can be of use, you can make it accept both, but item IDs are primarily numbers (for which we have constants) and we should never force scripters to use strings instead.